packages/server-core-integration/src/lib/gatewaySetup.ts

```diff
@@ -0,0 +1,215 @@
+import * as Winston from 'winston'
+
+/**
+ * Used when JSON.stringifying values that might be circular
+ * Usage: JSON.stringify(value, JSONStringifyCircular()))
+ */
+function JSONStringifyCircular() {
```

Comment:
To continue our talk from Tuesday.
A small thing one can do to improve the readability of the code a lot is to declare the return type so you don't have to read the implementation to figure it out.

This scenario highlights that since `JSONStringify` is a popuplar method and returns a String. Then when I see a method named `JSONStringifyCircular` I would also expect that that also returns a String, since the name of the method expands on the name of `JSONStringify`. Instead it returns `(key: string, value: any) => any`, which is vastly different from the return type of `JSONStringify`.

It seems small, but it means that a maintenance developer doesn't have to spend time reading the implementation of `JSONStringifyCircular` when said developer is trying to understand another function that uses `JSONStringifyCircular`.
And not having to go read the implementations of other methods all the time to figure the return type is a big time saver.

Reply:
We should consider enabling [@typescript-eslint/explicit-function-return-type](https://github.com/typescript-eslint/typescript-eslint/blob/v2.34.0/packages/eslint-plugin/docs/rules/explicit-function-return-type.md), afaIk it's already part of our code standard

Reply:
I would be for that :)

I would also suggest that we start adding the types to our variables more.
Offcourse if it's **obvious** what the type is, then it's not that necessary (but still nice).
So in a case like `const amountOfTries = 1` it's obvious it has the type number, so we can leave it out, but in a case like `const piece = fetchPiece()` it's not obvious what the type is (and what attributes the piece has, so it's nice to have the interface by hand).

Reply:
I agree that I would expect `JSONStringifyCircular` to return a string, and not be the parameter to pass to `JSON.stringify`.